s3/replication/build_and_test.sh

```diff
@@ -0,0 +1,198 @@
+
+# Replication directory
+REPLICATION_DIR="$(pwd)"
+echo "Replication directory :" $REPLICATION_DIR
+
+REPLICATION_VENV=$REPLICATION_DIR/s3env/bin
+echo "Virtual environment directory : "$REPLICATION_VENV
+
+# Module directory
+S3_REPLICATION_COMMON_SRC_DIR=$REPLICATION_DIR/common/src
+S3_REPLICATION_MANAGER_SRC_DIR=$REPLICATION_DIR/manager/src
+S3_REPLICATOR_SRC_DIR=$REPLICATION_DIR/replicator/src
+
+# Test directory
+S3_REPLICATION_COMMON_TEST_DIR=$REPLICATION_DIR/common/tests/system
+S3_REPLICATION_MANAGER_TEST_DIR=$REPLICATION_DIR/manager/tests/system
+S3_REPLICATOR_TEST_DIR=$REPLICATION_DIR/replicator/tests/system
+
+echo "Module directories -"
+echo "Common module directory :" $S3_REPLICATION_COMMON_SRC_DIR
+echo "Manager module directory :" $S3_REPLICATION_MANAGER_SRC_DIR
+echo "Replicator module directory :" $S3_REPLICATOR_SRC_DIR
+
+# Helper method to stop background services.
+stop_replication_services()
+{
+    systemctl stop s3_in_memory.service
+    systemctl stop s3replicationmanager.service
+    systemctl stop s3replicator.service
+
+    # Check if any service name is passed or not.
+    # If passed then it's an error case. Print an error message and exit.
+    if [ $# -eq 0 ]; then
+        echo "Shut down all processes!"
+    else
+        FAILED_SVC=$1
+        echo "ERROR : $FAILED_SVC service activation failed!"
+        exit -1
+    fi
+}
+
+# Active an environment
+python3 -m venv $REPLICATION_DIR/s3env
+source $REPLICATION_DIR/s3env/bin/activate
+
+# Check and download dependencies
+pip3 install --upgrade setuptools
+
+cd $REPLICATION_DIR
+make install
+
+# Create and copy files
+S3_IN_MEMRORY_SVC_FILE=$S3_REPLICATION_COMMON_SRC_DIR/systemd/s3_in_memory.service
+MANGER_SVC_FILE=$S3_REPLICATION_MANAGER_SRC_DIR/systemd/s3replicationmanager.service
+REPLICATOR_SVC_FILE=$S3_REPLICATOR_SRC_DIR/systemd/s3replicator.service
+
+# Systemd service files
+SYSD_S3_IN_MEMORY_SVC_FILE=/etc/systemd/system/s3_in_memory.service
+SYSD_REPLICATION_MANAGER_SVC_FILE=/etc/systemd/system/s3replicationmanager.service 
+SYSD_REPLICATOR_SVC_FILE=/etc/systemd/system/s3replicator.service
+
+# Create creadential file in home directory with below path.
+# ~/.cortxs3/credentials.yaml
+CORTX_S3_CRED_FILE=~/.cortxs3/credentials.yaml
+CORTX_S3_DIR="$(dirname "$CORTX_S3_CRED_FILE")"
+if [ -f "$CORTX_S3_CRED_FILE" ]; then
+    echo "$CORTX_S3_CRED_FILE exist!"
+else
+    if [ -d $CORTX_S3_DIR]; then
+        echo "$CORTX_S3_DIR exist!"
+    else
+        mkdir $CORTX_S3_DIR
+    fi
+    touch $CORTX_S3_CRED_FILE
+    # Placeholder for keys to run against in-memory s3 server.
+    echo "access_key: '<your access_key>'" > $CORTX_S3_CRED_FILE
+    echo "secret_key: '<your secret_key>'" >> $CORTX_S3_CRED_FILE
+fi
+
+rm -f $SYSD_S3_IN_MEMORY_SVC_FILE 
+rm -f $SYSD_REPLICATION_MANAGER_SVC_FILE
+rm -f $SYSD_REPLICATOR_SVC_FILE
+
+# Strip hardcoded lines
+head -n -1 $S3_IN_MEMRORY_SVC_FILE > $SYSD_S3_IN_MEMORY_SVC_FILE
```

Comment:
use placeholder key in service file as discussed and use sed to replace path.

Reply:
done